File: daily/Binary Tree/144-Binary-tree-preorder-traversal.go
package daily

func preorderTraversal(root *TreeNode) []int {
	var results []int
	dfsPreorder(root, &results)
	return results
}

func dfsPreorder(node *TreeNode, results *[]int) {
	if node == nil {
		return
	}
	*results = append(*results, node.Val)
	dfsPreorder(node.Left, results)
	dfsPreorder(node.Right, results)
}

File: daily/Binary Tree/144-Binary-tree-preorder-traversal.py
class Solution:
    def preorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        results = []
        def dfs(node):
            if not node:
                return
            results.append(node.val)
            dfs(node.left)
            dfs(node.right)
        dfs(root)
        return results